public/apps/configuration/app-router.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React from 'react';
+import { HashRouter as Router, Route, Switch } from 'react-router-dom';
+import { AppDependencies } from '../types';
+import { RouteItem } from './types';
+import { NavPanel } from './panels/nav-panel';
+import { RoleList } from './panels/role-list';
+import { SplitPage } from './pages/split-page';
+
+const RoutesMap: { [key: string]: RouteItem } = {
+  getStarted: {
+    name: 'Get Started',
+    href: '/getstarted',
+  },
+  roles: {
+    name: 'Roles',
+    href: '/roles',
+  },
+  users: {
+    name: 'Users',
+    href: '/internalusers',
+  },
+  permissions: {
+    name: 'Permissions',
+    href: '/permissions',
+  },
+  tenets: {
+    name: 'Tenets',
+    href: '/tenants',
+  },
+  auth: {
+    name: 'Authentication and authorization',
+    href: '/auth',
+  },
+};
+
+const RoutesList = [
+  RoutesMap.getStarted,
+  RoutesMap.roles,
+  RoutesMap.users,
+  RoutesMap.permissions,
+  RoutesMap.tenets,
+  RoutesMap.auth,
+];
+
+export function AppRouter(props: AppDependencies) {
+  return (
+    <Router basename={props.params.appBasePath}>
+      <Switch>
+        <Route path={RoutesMap.roles.href}>
+          <SplitPage
+            left={<NavPanel items={RoutesList} />}
+            right={<RoleList />}
+          />
```

Comment:
Trying to understand the thought process here.  Is this to ensure we always have sidebar on the all the pages right ?

We could take different and simplified approach. While defining router you could also add the side bar and for the other half of the page let's have actual content. You may completely avoid the split component. 

Some reference form other plugin

https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/166cf1c6acfc8eda1429bc1d516fe1fe0c8041a8/public/pages/main/Main.tsx#L69-L117

Reply:
The side nav is only for all the listing pages, not detail pages.
<img width="1431" alt="ss_role_detail" src="https://user-images.githubusercontent.com/63078162/79782909-a760ee00-82f4-11ea-899a-46eb690ac025.png">

Reply:
Handled with an regex patten route.